File: Python/subarray.py
def print_all_subarrays(arr):
    n = len(arr)
    print("All subarrays:")
    for start in range(n):
        for end in range(start, n):
            print(arr[start:end+1])

def main():
    n = int(input("Enter the number of elements in the array: "))
    print("Enter the elements separated by space:")
    arr = list(map(int, input().split()))
    
    if len(arr) != n:
        print("Warning: Number of elements entered does not match specified size.")
    
    print_all_subarrays(arr)

if __name__ == "__main__":
    main()

File: subarray.java
import java.util.Scanner;

public class subarray {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Read size of the array
        System.out.print("Enter the number of elements in the array: ");
        int n = sc.nextInt();

        int[] arr = new int[n];

        // Read array elements
        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        // Generate and print all subarrays
        System.out.println("All subarrays are:");
        for (int start = 0; start < n; start++) {
            for (int end = start; end < n; end++) {
                // Print elements from start to end index
                System.out.print("[");
                for (int k = start; k <= end; k++) {
                    System.out.print(arr[k]);
                    if (k < end) {
                        System.out.print(", ");
                    }
                }
                System.out.println("]");
            }
        }

        sc.close();
    }
}
